### PR TITLE
Enabled Stream/Consumer not recreated test in Eventing FI tests

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Enabled Stream/Consumer not recreated test in Eventing FI tests
- We disabled it because the job was flaky. https://github.com/kyma-project/kyma/issues/17146
- [Test Run](https://status.build.kyma-project.io/view/gs/kyma-prow-logs/pr-logs/pull/kyma-project_kyma/17312/mfaizanse_test_of_prowjob_pre-main-kyma-gardener-gcp-eventing-u/1646866079357603840)

**Related issue(s)**

- https://github.com/kyma-project/kyma/issues/17146

### CODE REVIEW

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 1]
```diff
@@ -223,52 +224,78 @@ describe('Eventing tests', function() {
...
-        if (!isUpgradeJob) {
-          debug(`Skipping saving JetStream test data into a configMap: ${jetStreamTestConfigMapName}`);
+  function jsTestStreamConsumerNotRecreatedTestSuite(upgradeStage='pre') {
+    // The test scenario is:
+    // 1. Before upgrade, save the stream and consumer creation timestamp in a configMap.
+    // 2. Upgrade the Kyma cluster.
+    // 3. After upgrade, fetch again the stream and consumer creation timestamp from NATS.
+    // 4. Compares the stream and consumer creation timestamp before and after upgrade.
+    // 5. If the timestamps are same, we conclude that the stream/consumer is not re-created.
+    // Note: It only checks consumer of first subscription (i.e. subscriptionsTypes[0]).
```
Comment (muralov) [inline tests/fast-integration/eventing-test/eventing-test.js:234]: 👍🏼

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 2]
```diff
@@ -223,52 +224,78 @@ describe('Eventing tests', function() {
...
 
-      it('upgrade should not have re-created consumers', async function() {
-        debug('Verifying that the consumers were not re-created by the kyma upgrade...');
-        await checkConsumerNotReCreated(natsApiRuleVSHost, preUpgradeConsumersData);
+        before('fetch data from configMap', async function() {
+          debug(`fetch configMap: ${jsRecreatedTestConfigMapName}...`);
+          const cm = await getConfigMapWithRetries(jsRecreatedTestConfigMapName, testNamespace);
+          if (!cm) {
+            // TODO: Remove this once these changes are migrated to release-branch
+            debug(`Skipping stream and consumers not re-created check because configMap was not found!`);
```
Comment (muralov) [inline tests/fast-integration/eventing-test/eventing-test.js:278]: 👍🏼

File: tests/fast-integration/eventing-test/eventing-test.js [hunk 3]
```diff
@@ -223,52 +224,78 @@ describe('Eventing tests', function() {
...
+            debug(`Skipping post-upgrade tasks...`);
+            this.skip();
+          }
+          debugBanner('Post-upgrade tasks to verify that stream and consumer not re-created check with NATS backend');
+        });
 
-      it('upgrade should not have re-created consumers', async function() {
-        debug('Verifying that the consumers were not re-created by the kyma upgrade...');
-        await checkConsumerNotReCreated(natsApiRuleVSHost, preUpgradeConsumersData);
+        before('fetch data from configMap', async function() {
```
Comment (muralov) [inline tests/fast-integration/eventing-test/eventing-test.js:273]: nitpick:  it is better to use `fetch data stream and consumer data from configMap`

File: tests/fast-integration/eventing-test/utils.js [hunk 4]
```diff
@@ -612,7 +617,7 @@ async function saveJetStreamDataForRecreateTest(host, configMapName) {
   };
 
   debug(`Saving fetched stream and consumers details in configMap (name: ${configMapName})...`);
-  await createK8sConfigMap(cmData, configMapName, testNamespace);
+  await createK8sConfigMapWithRetries(cmData, configMapName, testNamespace);
```
Comment (muralov) [inline tests/fast-integration/eventing-test/utils.js:620]: 👍🏼

**Comment by muralov** [comment]
Very good improvements! Only one nitpick. 
The upgrade job `pre-main-kyma-gardener-gcp-eventing-upgrade` is still failing.

File: tests/fast-integration/eventing-test/eventing-test.js (hunk 3 above)
Comment (mfaizanse) [reply tests/fast-integration/eventing-test/eventing-test.js]: updated

**Comment by mfaizanse** [comment]
> Very good improvements! Only one nitpick. The upgrade job `pre-main-kyma-gardener-gcp-eventing-upgrade` is still failing.

Well, the job is still flaky, now it failed because EventMesh subscription took more time to get ready. Nothing related to this PR.

**Review by muralov — APPROVED** [review]

**Pull request merged**